pkg/recipes/configloader/secrets.go

```diff
@@ -56,32 +60,49 @@ func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKe
 		}
 
 		// Populate the secretData map.
-		secretData, err = populateSecretData(secretStoreID, secretKeys, &secrets)
+		secretStoreData, err := populateSecretData(secretStoreID, secretKeysFilter, &secrets)
 		if err != nil {
 			return nil, err
 		}
+
+		// Merge the secret data for current secret store id into returned secretData map.
+		for key, value := range secretStoreData {
+			secretData[key] = value
+		}
 	}
 
 	return secretData, nil
 }
 
 // populateSecretData is a helper function to populate secret data from a secret store.
-func populateSecretData(secretStoreID string, secretKeys []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
-	secretData := make(map[string]map[string]string)
-
+// It takes a secret store ID, a filter for secret keys, and a response containing the secrets.
+// If the secret keys filter is nil or empty, it retrieves data for all keys in the secret store.
+// It returns a map where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
+func populateSecretData(secretStoreID string, secretKeysFilter []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
 	if secrets == nil {
 		return nil, fmt.Errorf("secrets not found for secret store ID '%s'", secretStoreID)
 	}
 
-	for _, secretKey := range secretKeys {
-		if secretDataValue, ok := secrets.Data[secretKey]; ok {
-			if secretData[secretStoreID] == nil {
-				secretData[secretStoreID] = make(map[string]string)
-			}
-			secretData[secretStoreID][secretKey] = *secretDataValue.Value
-		} else {
+	secretData := make(map[string]map[string]string)
+
+	// If secretKeys is nil or empty, retrieve data for all keys
+	if len(secretKeysFilter) == 0 {
+		secretKeysFilter = make([]string, 0, len(secrets.Data))
+		for secretKey := range secrets.Data {
+			secretKeysFilter = append(secretKeysFilter, secretKey)
+		}
+	}
+
+	for _, secretKey := range secretKeysFilter {
```

Comment:
i think this should be part of `else` block?

Reply:
this is correct as is. if condition ln89 populates secretKeysFilter with data for all keys if empty as described.
the for loop ln96 populates the return type for the list of keys.